src/sqlfluff/rules/L007.py

```diff
@@ -58,51 +68,103 @@ def _eval(self, context: RuleContext) -> LintResult:
+        results: List[LintResult] = []
+        # If len(operator_segments) == 0 this will essentially not run
+        for operator in operator_segments:
+            start = expr.reversed().first(FirstCodeAfterOperator(operator))
+            end = expr.first(FirstCodeAfterOperator(operator))
+            res = [
+                expr.select(start_seg=start.get(), stop_seg=operator),
+                expr.select(start_seg=operator, stop_seg=end.get()),
+            ]
+            # anchor and change els are reversed in the before case
+            if operator_new_lines == "before":
+                res = [els.reversed() for els in reversed(res)]
+
+            change_list, anchor_list = res
+            # If the anchor side of the list has no newline
+            # then everything is ok already
+            if not anchor_list.any(sp.is_name("newline")):
+                continue
+
+            insert_anchor = anchor_list.last().get()
+            assert insert_anchor, "Insert Anchor must be present"
+            lint_res = _generate_fixes(
+                operator_new_lines,
+                change_list,
+                operator,
+                insert_anchor,
+            )
+            results.append(lint_res)
+
+        if len(results) == 0:
+            return None
+        return results
+
+
+class FirstCodeAfterOperator:
+    """Predicate Fn Class."""
+
+    def __init__(self, operator: BaseSegment) -> None:
+        """Predicate with some memory of what it has seen."""
+        self.has_seen_operator = False
+        self.operator = operator
+
+    def __call__(self, seg: BaseSegment) -> bool:
+        """Call the predicate as per its creation."""
+        if seg == self.operator:
+            self.has_seen_operator = True
+            return False
+
+        if not self.has_seen_operator:
+            return False
+
+        if seg.is_code:
+            return True
+
+        return False
+
+
+def _generate_fixes(
+    operator_new_lines: str,
+    change_list: Segments,
+    operator: BaseSegment,
+    insert_anchor: BaseSegment,
+) -> LintResult:
+    # Duplicate the change list and append the operator
+    inserts: List[BaseSegment] = [
+        *change_list,
+        operator,
+    ]
+
+    if operator_new_lines == "before":
+        # We do yet another reverse here,
+        # This could be avoided but makes all "changes" relate to "before" config state
+        inserts = list(reversed(inserts))
+
+    # ensure to insert in the right place
+    edit_type = "create_before" if operator_new_lines == "before" else "create_after"
+    fixes = [
+        # Insert elements reversed
+        LintFix(
+            edit_type=edit_type,
+            edit=list(map(lambda el: copy.deepcopy(el), reversed(inserts))),
+            anchor=insert_anchor,
+        ),
+        # remove the Op
+        LintFix.delete(operator),
+        # Delete the original elements (related to insert)
+        *[LintFix.delete(seg) for seg in change_list],
+    ]
```

Comment:
I wonder if this would be easier to read?
```suggestion
    ] + [
        LintFix.delete(seg) for seg in change_list
    ]
```

Reply:
Black was giving me huge issues with complaining about various linting in and round these style things

Reply:
Yea nah - I tried and black wants it on one line if you go for a + operator. and then the comment is floating in the wrong place

Reply:
What about
`*change_list.apply(LintFix.delete),`

Reply:
`apply()` is good. Don't worry too much about this; I was just curious -- don't want to obsess **too much** over style. How much is "just enough", ha ha? 🤔